In file included from cm_device.cpp:31:
In file included from ./cm_device.h:37:
In file included from ./cm_array.h:31:
./cm_mem.h:200:17: error: invalid output size for constraint '=a'
                                             :"=a"(local_rax), "=r"(local_rbx),
                                                   ^

--- src/cm_mem.h.orig	2016-09-07 23:51:38 UTC
+++ src/cm_mem.h
@@ -194,7 +194,9 @@ inline void GetCPUID(int CPUInfo[4], int InfoType)
 			CPUInfo[1] = local_ebx;
 			CPUInfo[2] = local_ecx;
 			CPUInfo[3] = local_edx;
-		} else {
+		}
+#ifdef __LP64__
+		  else {
 			uint64_t local_rax, local_rbx, local_rcx, local_rdx;
 			__asm__ __volatile__("push %%rbx      \n\t" "cpuid           \n\t" "mov %%rbx, %1   \n\t" "pop %%rbx       \n\t"	/* restore the old %ebx */
 					     :"=a"(local_rax), "=r"(local_rbx),
@@ -206,6 +208,7 @@ inline void GetCPUID(int CPUInfo[4], int InfoType)
 			CPUInfo[2] = local_rcx;
 			CPUInfo[3] = local_rdx;
 		}
+#endif
 
 #ifndef NO_EXCEPTION_HANDLING
 	}
